src/mlpack/core/data/load.hpp

```diff
@@ -287,12 +288,33 @@ bool Load(const std::string& filename,
           const bool fatal = false,
           format f = format::autodetect);
 
+/**
+ * Image load/save interfaces.
+ */
+#ifdef HAS_STB
+template<typename eT>
+bool Load(const std::string& filename,
+          arma::Mat<eT>& matrix,
+          ImageInfo& info,
+          const bool fatal = false,
+          const bool transpose = true);
+
+template<typename eT>
+bool Load(const std::vector<std::string>& files,
+          arma::Mat<eT>& matrix,
+          ImageInfo& info,
+          const bool fatal = false,
+          const bool transpose = true);
+#endif // HAS_STB.
```

Comment:
It could be really useful to also have something like

```
#else 
struct ImageInfo { };

template<typename eT>
bool Load(const std::string& filename,
          arma::Mat<eT>& matrix,
          ImageInfo& info,
          const bool fatal = false,
          const bool transpose = true)
{
  throw std::runtime_error("Load(): HAS_STB is not defined, so STB is not available and images cannot be loaded!");
}

template<typename eT>
bool Load(const std::vector<std::string>& files,
          arma::Mat<eT>& matrix,
          ImageInfo& info,
          const bool fatal = false,
          const bool transpose = true)
{
  throw std::runtime_error("Load(): HAS_STB is not defined, so STB is not available and images cannot be loaded!");
}
```

Also it would be useful to add documentation to the top of the definitions here, so that Doxygen will pick it up.